kong/llm/drivers/shared.lua

```diff
@@ -250,19 +318,48 @@ function _M.http_request(url, body, method, headers, http_opts)
     httpc:set_proxy_options(http_opts.proxy_opts)
   end
 
-  local res, err = httpc:request_uri(
-    url,
-    {
-      method = method,
-      body = body,
-      headers = headers,
+  local parsed = parse_url(url)
+
+  if buffered then
+    local ok, err, _ = httpc:connect({
+      scheme = parsed.scheme,
+      host = parsed.host,
+      port = parsed.port or 443,  -- this always fails. experience.
```

Comment:
> this always fails. experience.

Does this mean that `parsed.port` is always `nil`, or does it mean something else?

If the former, I have no idea if this would fix your problem, but you could try out the [parse_uri function in resty.http](https://github.com/ledgetech/lua-resty-http/tree/master?tab=readme-ov-file#parse_uri) to see if it works any better.

Reply:
Unfortunately if user doesn't explicitly put `:443` in their URL, it doesn't get parsed.

This isn't anything I can really fix, because it is handled automatically by our typedefs package, and I don't want to break any other plugins for this one change.

I think it's safer just to assume 443 is default port, unless otherwise told...?